### PR TITLE
tests: move to pytest

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -56,3 +56,6 @@ ChangeLog
 # generated docs
 doc/source/api
 doc/source/gnocchi.rst
```
Comment (jd) [inline .gitignore]: remove .testrepository too

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 15f585478bd6d3f6b529547771a4355410e1c8c0

**Comment by sileht** [comment]
Pull-request updated, HEAD is now b4091b398e499139fe20739ed27522e36b715a30

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 73114ca1c0e812f5204d4f70c8eddda6e8ef708f

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 36cc11ade1996deefce94e10eda07e7a1a3bcf67

File: setup.cfg [hunk 2]
```diff
@@ -79,9 +79,10 @@ openstack.metric.v1 =
 [extras]
 test =
   coverage>=3.6
```
Comment (jd) [inline setup.cfg]: Sounds like we can remove that if we depend on pytest-cov

File: setup.cfg [hunk 3]
```diff
@@ -79,9 +79,10 @@ openstack.metric.v1 =
...
-  python-subunit>=0.0.18
-  testrepository>=0.0.18
   testtools>=1.4.0
+  pytest
+  pytest-cov
```
Comment (jd) [inline setup.cfg]: Maybe only add in the cover target? That's the only place someone would need it.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now a1ac85b712d308ceca06e61270e0ee491ad3270a

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 5669eafb108e1504f05781ed7b90067ba4515e9d

File: setup.cfg (hunk 3 above)
Comment (sileht) [reply setup.cfg]: done

File: setup.cfg (hunk 2 above)
Comment (sileht) [reply setup.cfg]: done

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 1086ac4c5619e82253417870aaf8a1275194d756

**Pull request merged**